### ISSUE TITLE
Support Redis connection-defined prefix

### ISSUE THREAD

**Issue by iksaku** [DirectoryTree/ActiveRedis#3]
While testing, I noticed that any `SCAN` commands were not really matching the keys available in the DB. After digging a little, I found this issue in the Laravel repo: https://github.com/laravel/framework/issues/24257. This illustrates the issue at hand were `SCAN` commands executed when using the phpredis extension, do not prepend the prefix set in the client by Laravel.

An issue was raised to the phpredis later on, and they added an opt-in configuration to solve this issue: https://github.com/phpredis/phpredis/issues/548.

The solution is to add the following somewhere in the app (Service provider?):
```php
use Illuminate\Support\Facades\Redis as RedisConnection;
use Redis;

RedisConnection::setOption(Redis::OPT_SCAN, Redis::SCAN_PREFIX);
```

I hope this is useful for documentation, but I think it would be best if we could have this done in the package service provider for us.

Have a great day!

**Comment by stevebauman** [comment]
Hey @iksaku, thanks so much for trying ActiveRedis! I appreciate the detailed issue here. I will definitely add this into the repo 🙏 

Will close this out once added 👍

**Comment by iksaku** [comment]
While my solution solves the issue with scanning, it also prevents other attributes from bing pulled properly.

This is because phpredis `SCAN` returns keys with prefixes and then we try to query keys with such prefix already added, and then phpredis re-prefixes. Seems like normalizing such keys after a scan leads to the expected results:

```diff
public function chunk(string $pattern, int $count): Generator
    {
        $cursor = null;

        do {
            [$cursor, $keys] = $this->redis->scan($cursor, [
                'match' => $pattern,
                'count' => $count,
            ]);

            if (is_null($keys)) {
                return;
            }

            if (! empty($keys)) {
+                if ($this->redis instanceof PhpRedisConnection) {
+                    $keys = array_map(
+                        fn (string $key) => Str::after($key, $this->redis->getOption(\Redis::OPT_PREFIX)),
+                        $keys
+                    );
+                }

                yield $keys;
            }
        } while ($cursor !== '0');
    }
```

**Comment by iksaku** [comment]
Looking a bit more into this. It looks like Predis is also affected as long as the Laravel configuration has a prefix set (which is the default).

Both PhpRedis and Predis seem to behave the same way:
* They skip prefixing on pattern provided to `SCAN`.
* Upon matches from `SCAN`, neither remove connection-defined prefix from the resulting keys, causing calls to `Repository::getAttributes()` double prefix keys and, you guessed it, not finding the corresponding keys.

Seems like this is an issue with the library not supporting prefixes at all.

I've been digging a lot into this, so instead I'll open a PR with a simple fix I found 🤓

**Comment by stevebauman** [comment]
Released [v1.4.1](https://github.com/DirectoryTree/ActiveRedis/releases/tag/v1.4.1) with your PR in it, thanks for your work! 🙏
